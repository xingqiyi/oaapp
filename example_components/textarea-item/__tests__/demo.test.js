import { webDemoTest } from '../../../tests/shared/demoTest';

webDemoTest('textarea-item');
