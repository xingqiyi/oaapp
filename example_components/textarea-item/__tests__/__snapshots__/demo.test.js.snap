// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/textarea-item/demo/basic.md correctly 1`] = `
<div>
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Customize to focus
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          标题
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            data-seed="logId"
            placeholder="auto focus in Alipay client"
            rows="1"
          />
        </div>
      </div>
      <div
        class="am-list-item am-textarea-item"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          标题
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            data-seed="logId"
            placeholder="click the button below to focus"
            rows="1"
          />
        </div>
      </div>
      <div
        class="am-list-item am-list-item-middle"
      >
        <div
          class="am-list-line"
        >
          <div
            class="am-list-content"
          >
            <div
              style="width:100%;color:#108ee9;text-align:center;"
            >
              click to focus
            </div>
          </div>
        </div>
        <div
          class="am-list-ripple"
          style="display:none;"
        />
      </div>
    </div>
  </div>
  <div
    class="am-whitespace am-whitespace-md"
  />
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Whether is controlled
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item am-textarea-item-single-line"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          受控组件
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder="Hello World"
            rows="1"
          />
        </div>
      </div>
      <div
        class="am-list-item am-textarea-item am-textarea-item-single-line"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          非受控组件
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder="please input content"
            rows="1"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Auto / Fixed height
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          高度自适应
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder=""
            rows="1"
          />
        </div>
      </div>
      <div
        class="am-list-item am-textarea-item"
      >
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder="fixed number of lines"
            rows="3"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="am-whitespace am-whitespace-md"
  />
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Show clear icon
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item am-textarea-item-single-line"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          标题
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder="displayed clear icon while typing"
            rows="1"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="am-whitespace am-whitespace-md"
  />
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Custom title（text / image / empty）
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item am-textarea-item-single-line"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          <img
            alt="icon"
            src="https://os.alipayobjects.com/rmsportal/mOoPurdIfmcuqtr.png"
            style="width:0.56rem;height:0.56rem;"
          />
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder="title can be customized"
            rows="1"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="am-whitespace am-whitespace-md"
  />
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Limited value length
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item am-textarea-item-single-line"
      >
        <div
          class="am-textarea-control"
        >
          <textarea
            maxlength="10"
            placeholder="can enter up to 10 characters"
            rows="1"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="am-whitespace am-whitespace-md"
  />
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Count
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item"
      >
        <div
          class="am-textarea-control"
        >
          <textarea
            maxlength="100"
            placeholder=""
            rows="5"
          >
            计数功能,我的意见是...
          </textarea>
        </div>
        <span
          class="am-textarea-count"
        >
          <span>
            13
          </span>
          /100
        </span>
      </div>
    </div>
  </div>
  <div
    class="am-whitespace am-whitespace-md"
  />
  <div
    class="am-list"
  >
    <div
      class="am-list-header"
    >
      Not editable / Disabled
    </div>
    <div
      class="am-list-body"
    >
      <div
        class="am-list-item am-textarea-item am-textarea-item-single-line"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          姓名
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            placeholder=""
            readonly=""
            rows="1"
          >
            not editable
          </textarea>
        </div>
      </div>
      <div
        class="am-list-item am-textarea-item am-textarea-disabled am-textarea-item-single-line"
      >
        <div
          class="am-textarea-label am-textarea-label-5"
        >
          姓名
        </div>
        <div
          class="am-textarea-control"
        >
          <textarea
            disabled=""
            placeholder=""
            rows="1"
          >
            disabled style
          </textarea>
        </div>
      </div>
    </div>
  </div>
</div>
`;
